stats/influxdb/config.go

```diff
@@ -131,3 +139,9 @@ func (c *Config) UnmarshalJSON(data []byte) error {
 func (c Config) MarshalJSON() ([]byte, error) {
 	return json.Marshal(ConfigFields(c))
 }
+
+func (c *Config) defaultValues() {
```

Comment:
Hmm I'm not sure that doing this in the `UnmarshalText()` and `UnmarshalJSON()` (and actually having a `defaultValues()` function that changes a receiver's property) is very idiomatic and potentially it could be confusing to Go developers. A much more idiomatic solution would be to create a "constructor" function called `NewConfig()` that instantiates the config with the default values: https://hassansin.github.io/golang-constructor-like-function-use-cases  

This would also solve the issues with the suboptimal tests. The overhead is that elsewhere in the code you'd have to keep in mind that you'd have to use `NewConfig()` instead of directly instantiating a `Config` struct or, in those cases:
- https://github.com/loadimpact/k6/blob/master/cmd/login_influxdb.go#L49
- https://github.com/loadimpact/k6/blob/master/cmd/collectors.go#L71

and others like them, something like:
```go
conf := influxdb.NewConfig().Apply(config.Collectors.InfluxDB)
```

Reply:
I agree, it doesn't feel right as is. I'll take a look at changing this to a NewConfig() function.